internal/providers/openstack/openstack.go

```diff
@@ -44,11 +45,16 @@ const (
 )
 
 var (
-	metadataServiceUrl = url.URL{
+	metadataServiceUrlIPv4 = url.URL{
 		Scheme: "http",
 		Host:   "169.254.169.254",
 		Path:   "openstack/latest/user_data",
 	}
+	metadataServiceUrlIPv6 = url.URL{
+		Scheme: "http",
+		Host:   "fe80::a9fe:a9fe%",
```

Comment:
Have you tried having the host be [fe90::a9fe:a9fe%]? since you are getting an error on fetch due to improper escaping ?

Reply:
I agree with @prestist . Looks like you're missing to enclose the host in square brackets. 
At least it's documented [here](https://pkg.go.dev/net/url#URL).
Maybe you will also need to add the interface name to the host once you have it, before calling `FetchToBuffer`